sdks/java/io/tika/src/main/java/org/apache/beam/sdk/io/tika/TikaIO.java

```diff
@@ -0,0 +1,307 @@
+import org.apache.tika.metadata.Metadata;
+
+
+/**
+ * {@link PTransform} for parsing arbitrary files using Apache Tika.
+ * Files in many well known text, binary or scientific formats can be processed.
+ *
+ * <p>To read a {@link PCollection} from one or more files
+ * use {@link TikaIO.Read#from(String)}
+ * to specify the path of the file(s) to be read.
+ *
+ * <p>{@link TikaIO.Read} returns a bounded {@link PCollection} of {@link String Strings},
+ * each corresponding to a sequence of characters reported by Apache Tika SAX Parser.
+ *
+ * <p>Example:
+ *
+ * <pre>{@code
+ * Pipeline p = ...;
+ *
+ * // A simple Read of a local PDF file (only runs locally):
+ * PCollection<String> content = p.apply(TikaInput.from("/local/path/to/file.pdf"));
+ * }</pre>
+ */
+public class TikaIO {
+
+  /**
+   * A {@link PTransform} that parses one or more files and returns a bounded {@link PCollection}
+   * containing one element for each sequence of characters reported by Apache Tika SAX Parser.
+   */
+   public static Read read() {
+     return new AutoValue_TikaIO_Read.Builder()
+        .setQueuePollTime(Read.DEFAULT_QUEUE_POLL_TIME)
+        .setQueueMaxPollTime(Read.DEFAULT_QUEUE_MAX_POLL_TIME)
+        .build();
+   }
+
+   /** Implementation of {@link #read}. */
+  @AutoValue
+  public abstract static class Read extends PTransform<PBegin, PCollection<String>> {
+    private static final long serialVersionUID = 2198301984784351829L;
+    public static final long DEFAULT_QUEUE_POLL_TIME = 50L;
+    public static final long DEFAULT_QUEUE_MAX_POLL_TIME = 3000L;
+
+    @Nullable abstract ValueProvider<String> getFilepattern();
+    @Nullable abstract ValueProvider<String> getTikaConfigPath();
+    @Nullable abstract Metadata getInputMetadata();
+    @Nullable abstract Boolean getReadOutputMetadata();
+    @Nullable abstract Long getQueuePollTime();
+    @Nullable abstract Long getQueueMaxPollTime();
+    @Nullable abstract Integer getMinimumTextLength();
+    @Nullable abstract Boolean getParseSynchronously();
+
+    abstract Builder toBuilder();
+
+    @AutoValue.Builder
+    abstract static class Builder {
+      abstract Builder setFilepattern(ValueProvider<String> filepattern);
+      abstract Builder setTikaConfigPath(ValueProvider<String> tikaConfigPath);
+      abstract Builder setInputMetadata(Metadata metadata);
+      abstract Builder setReadOutputMetadata(Boolean value);
+      abstract Builder setQueuePollTime(Long value);
+      abstract Builder setQueueMaxPollTime(Long value);
+      abstract Builder setMinimumTextLength(Integer value);
+      abstract Builder setParseSynchronously(Boolean value);
+
+      abstract Read build();
+    }
+
+    /**
+     * A {@link PTransform} that parses one or more files with the given filename
+     * or filename pattern and returns a bounded {@link PCollection} containing
+     * one element for each sequence of characters reported by Apache Tika SAX Parser.
```

Comment:
This doesn't seem true - it is returning key"="value strings.

Reply:
As I noted above - only for the optional file metadata. May be I should update the docs to clarify it